WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/Downloadable Files/FIle List/ProductDownloadListViewModelTests.swift

```diff
@@ -0,0 +1,163 @@
+import XCTest
+@testable import WooCommerce
+@testable import Yosemite
+
+final class ProductDownloadListViewModelTests: XCTestCase {
+
+    // MARK: - Initialization
+
+    func testReadonlyValuesAreAsExpectedAfterInitializingAProductWithNonEmptyDownloadableFiles() throws {
+        // Arrange
+        let product = MockProduct().product(downloadable: true)
+        let model = EditableProductModel(product: product)
+
+        // Act
+        let viewModel = ProductDownloadListViewModel(product: model)
+
+        // Assert
+        XCTAssertEqual(viewModel.count(), 3)
+        XCTAssertEqual(viewModel.downloadLimit, 1)
+        XCTAssertEqual(viewModel.downloadExpiry, 1)
+        let file = try XCTUnwrap(viewModel.item(at: 0))
+        XCTAssertEqual(file.downloadableFile.downloadID, "1f9c11f99ceba63d4403c03bd5391b11")
+        XCTAssertEqual(file.downloadableFile.name, "Song #1")
+        XCTAssertEqual(file.downloadableFile.fileURL, "https://example.com/woo-single-1.ogg")
+    }
+
+    func testReadonlyValuesAreAsExpectedAfterInitializingAProductWithEmptyDownloadableFiles() {
+        // Arrange
+        let product = MockProduct().product(downloadable: false)
+        let model = EditableProductModel(product: product)
+
+        // Act
+        let viewModel = ProductDownloadListViewModel(product: model)
+
+        // Assert
+        XCTAssertEqual(viewModel.count(), 0)
+        XCTAssertEqual(viewModel.downloadLimit, -1)
+        XCTAssertEqual(viewModel.downloadExpiry, -1)
+    }
+
+    // MARK: - `handleDownloadableFilesChange`
+
+    func testHandlingADuplicateDownloadableFilesUpdatesWithError() {
+
+    }
+
+    func testHandlingAValidDownloadableFilesUpdatesWithSuccess() {
+
+    }
+
+    func testHandlingTheOriginalDownloadableFilesIsAlwaysValidAndUnique() {
+
+    }
+
+    func testHandlingAValidDownloadableFilesAddsWithSuccess() {
+
+    }
+
+    func testHandlingAValidExistingDownloadableFileRemovesWithSuccess() {
+
+    }
```

Comment:
Those are intentionally left empty. Those are related to adding/removing downloadable files. But the functionality hasn't been implemented yet. Which is due to be implemented with #2758 and #2759 . So I left those empty for now, to implement with the respective tickets. But I think It will be better to remove them and add back again with those tickets.